plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -45,11 +46,30 @@ func init() {
 type lifecycle struct {
 	*admission.Handler
 	client             clientset.Interface
-	store              cache.Store
 	immortalNamespaces sets.String
+	namespaceInformer  framework.SharedIndexInformer
+	// forceLiveLookupCache holds a list of entries for namespaces that we have a strong reason to believe are stale in our local cache.
+	// if a namespace is in this cache, then we will ignore our local state and always fetch latest from api server.
+	forceLiveLookupCache *lru.Cache
```

Comment:
> It's not necessary deleted. It could either be terminating or deleted. I think I will just keep force live lookups for now.

Somehow I've missed that.  It looked like you only called `.Add` during a Delete.  can you link me to where else you're adding it?